camel/agents/task_agent.py

```diff
@@ -181,3 +183,192 @@ def run(
+    ) -> None:
+
+        self.task_creator_prompt = TextPrompt(
+            """Create new tasks with the following objective: {objective}.
+The last completed task has the result:
+{task_result}.
+This result was based on this task: {task}.
+Based on the result, return a list of tasks to be completed in order to meet the objective.
+Return one task per line in your response. The result must be a numbered list in the format:
+
+#. First task
+#. Second task
+
+The number of each entry must be followed by a period. Be concise.
+{unsolved_tasks}.""")
+
+        system_message = BaseMessage(
+            role_name="Task Creator",
+            role_type=RoleType.ASSISTANT,
+            meta_dict=None,
+            content="You are a helpful task creator.",
+        )
+
+        super().__init__(system_message, model, model_config,
+                         output_language=output_language)
+
+        if (objective):
+            self.objective = objective
+            self.task_creator_prompt = self.task_creator_prompt.format(
+                objective=self.objective)
+
+    def run(
+        self,
+        previous_task: Union[str, TextPrompt],
+        task_result: Union[str, TextPrompt],
+        task_list: Optional[List[str]] = None,
+    ) -> List[str]:
+        r"""Generate subtasks based on the previous task results and
+        incomplete task list.
+
+        Args:
+            previous_task (Union[str, TextPrompt]): The last completed task to
+                be used to create future plans.
+            task_result (Union[str, TextPrompt]): The result of last completed
+                task to be used to create future plans.
+            task_list (List[str], optional): The incompleted task list
+                which should not overlap with new created tasks.
+        Returns:
+            List[str]: The new task list generated by the Agent.
+        """
+        self.reset()
+        self.task_creator_prompt = self.task_creator_prompt.format(
+            task=previous_task, task_result=task_result)
+        if task_list is not None:
+            unsolved = f"These are unsolved tasks: {task_list}.\n\
+                These new tasks must not overlap with incomplete tasks."
+
+            self.task_creator_prompt = self.task_creator_prompt.format(
+                unsolved_tasks=unsolved)
+        else:
+            self.task_creator_prompt = self.task_creator_prompt.format(
+                unsolved_tasks="")
+
+        task_msg = BaseMessage.make_user_message(
+            role_name="Task Planner", content=self.task_creator_prompt)
+
+        task_response = self.step(task_msg)
+
+        if task_response.msgs is None:
+            raise RuntimeError("Got None Subtasks messages.")
+        if task_response.terminated:
+            raise RuntimeError("Task planning failed.")
+
+        sub_tasks_msg = task_response.msgs[0]
+        new_tasks = sub_tasks_msg.content.strip().split('\n')
+        return get_task_list(new_tasks)
```

Comment:
Let's move `*.strip().split('\n')` inside the `get_task_list` helper function.
```suggestion
        return get_task_list(sub_tasks_msg.content)
```